libs/blocks/bulk/bulk.js

```diff
@@ -0,0 +1,345 @@
+
+// eslint-disable-next-line max-len
+// TODO: enable IMS sign in when the code is merged on the main branch, as IMS does not work on the PR branch
+const IMS_SIGN_IN_ENABLED = false;
+const URLS_ENTRY_LIMIT = 1000;
+
+function User({ user }) {
+  return html`
+      <div class="bulk-user">
+          <div class="bulk-user-header">logged in as</div>
+          <div class="bulk-user-name">
+              ${user.name}
+          </div>
+          <a class="bulk-user-signout" onclick=${signOut}>Sign out</a>
+      </div>
+  `;
+}
+
+function UrlInput({ urlsElt }) {
+  return html`
+    <span class="max-urls">Maximum number of URLS processed: ${URLS_ENTRY_LIMIT} </span>
+    <textarea class="bulk-urls-input" placeholder="Ex: https://main--milo--adobecom.hlx.page/my/sample/page" ref="${urlsElt}">${getStoredUrlInput()}</textarea>
+  `;
+}
+
+function SelectBtn({ actionElt, onSelectChange, storedOperationName }) {
+  return html`
+      <select class="bulk-action-select" ref="${actionElt}" onChange=${onSelectChange}>
+          <option value="preview">Preview</option>
+          <option value="publish" selected="${storedOperationName === 'publish'}">Publish</option>
+          <option value="preview&publish" selected="${storedOperationName === 'preview&publish'}">Preview & Publish</option>
+      </select>
+  `;
+}
+
+function SubmitBtn({ submit }) {
+  return html`
+    <button class="bulk-action-submit" onClick=${submit}>
+      Run process
+    </button>
+  `;
+}
+
+function prettyDate() {
+  const date = new Date();
+  const localeDate = date.toLocaleString();
+  const splitDate = localeDate.split(', ');
+  return html`
+    <span class=bulk-date>${splitDate[0]}</span>
+    <span class=bulk-time>${splitDate[1]}</span>
+  `;
+}
+
+function bulkPublishStatus(row) {
+  const status = row.status.publish !== 200
+    ? `Error  - Status: ${row.status.publish}`
+    : '';
+  return row.status.publish !== 200 && row.status.publish !== undefined && html`
+    <span class=page-status>${status}</span>
+  `;
+}
+
+function bulkPreviewStatus(row) {
+  const status = row.status.preview !== 200
+    ? `Error - Status: ${row.status.preview}`
+    : '';
+  return row.status.preview !== 200 && row.status.preview !== undefined && html`
+    <span class=page-status>${status}</span>
+  `;
+}
+
+function StatusTitle({ bulkTriggered, submittedAction, urlNumber }) {
+  const name = getActionName(submittedAction, true);
+  const URLS = (urlNumber > 1) ? 'URLS' : 'URL';
+  return bulkTriggered && html`
+    <div class="bulk-status-head">
+      <ul class="bulk-status-head-title">
+        <li><span>STATUS:</span> ${name} ${urlNumber} ${URLS}</li>
+      </ul>
+    </div>
+  `;
+}
+
+function StatusRow({ row }) {
+  const timeStamp = prettyDate();
+  const errorStyle = 'status-error';
+  const previewStatusError = row.status.preview !== 200
+    ? errorStyle
+    : '';
+  const publishStatusError = row.status.publish !== 200
+    ? errorStyle
+    : '';
```

Comment:
I'm always confused by having negative ternaries - it's clearer to swap the conditions around, something like:
```suggestion
  const previewStatusError = row.status.preview === 200 ? '' : errorStyle
  const publishStatusError = row.status.publish === 200 ? '' : errorStyle
```

Reply:
Addressed in https://github.com/adobecom/milo/pull/559/commits/3c8cc9387ed85003d8590024063603cba69b2b11